packages/core-plugin-api/src/plugin/types.ts

```diff
@@ -44,6 +44,17 @@ export type AnyRoutes = { [name: string]: RouteRef | SubRouteRef };
  */
 export type AnyExternalRoutes = { [name: string]: ExternalRouteRef };
 
+/**
+ * Catch-all metadata type.
+ *
+ * @public
+ */
+export type AnyPluginOptions = { [name: string]: any };
```

Comment:
```suggestion
export type AnyPluginOptions = { [name: string]: unknown };
```

To make sure consumers of the type don't shoot themselves in the foot